x/poe/contract/tgrade_valset.go

```diff
@@ -402,6 +401,31 @@ func (v ValsetContractAdapter) ListValidators(ctx sdk.Context, pagination *Pagin
 	return vals, cursor, nil
 }
 
+// IterateActiveValidators iterate through the active validator set.
+func (v ValsetContractAdapter) IterateActiveValidators(ctx sdk.Context, callback func(ValidatorInfo) bool) error {
+	var rsp ListActiveValidatorsResponse
+	cursor, err := v.doPageableQuery(ctx, ValsetQuery{ListActiveValidators: &ListValidatorsQuery{}}, &rsp)
+	if err != nil {
+		return err
+	}
+	for !cursor.Empty() {
+		for _, v := range rsp.Validators {
+			if callback(v) {
+				return nil
+			}
+		}
+		newCursor, err := v.doPageableQuery(ctx, ValsetQuery{ListActiveValidators: &ListValidatorsQuery{StartAfter: cursor.String()}}, &rsp)
```

Comment:
Ah, this is the wrong start after. 

It should be `rsp.Validators[len(rsp.Validators)-1].Address` (not 100% on the name of the field, but the address of the last validator returned).

Reply:
Or does the underlying contract not support pagination?

Reply:
@maurolacy can you double check my code please?

Reply:
Taking a look now.

Reply:
This is correct AFAIK. It's a little tricky with the pagination cursor, but this code is similar / equivalent to the one in `TestListAllValidatorsViaCursor` integration test (it just uses the `doPageableQuery` wrapper).

The cursor is a copy of some of the last elements of the page (the operator address).